src/utils/hex.ts

```diff
@@ -25,9 +29,10 @@ export function padString(
  * @param length - The final length in bytes of the hex string.
  */
 export function padBigint(
-  b: bigint,
+  b: bigint | undefined,
   length: number,
 ): PrefixedHexString {
+  b = b ?? 0n;
```

Comment:
I'm surprised this works!

I think we should just put this inside line 36:
return "0x" + (stripHexPrefix(bigIntToHex(b ?? 0n)).padStart(2 * length, "0"));